### PR TITLE
Add summary report showing overall view from faceted projects

### PR DESCRIPTION
**Pull request by cookkate**
In progress.

This report summarizes complex experiments with multiple variables (e.g. dose + timepoint) for which it's difficult to get an overall view using the faceted reports.

TODO:

- [x] Add headers
- [ ] Flesh out description of each plot
- [ ] fix the legend text for heatmaps (that currently all say "matrixXX"
- [ ] test with another dataset with categorical variables

### CODE REVIEW

**Comment by mattjmeier** [comment]
Also add an option in the config (true or false) to write the report or not

**Comment by cookkate** [comment]
Ok I tried a few things to try and fix the GO report with no success, so I reverted my commits.

File: scripts/data_functions.R [hunk 1]
```diff
@@ -70,9 +70,9 @@ sort_contrasts <- function(exp_metadata, contrasts, design, sortcol){
 }
 
 process_data_and_metadata <- function(count_data, exp_metadata, contrasts, intgroup, design, params){
-    count_data <- filter_data(count_data, exp_metadata, params$nmr_threshold)
     exp_metadata <- filter_metadata(exp_metadata, params, design)
     exp_metadata <- format_and_sort_metadata(exp_metadata, intgroup, design, params$sortcol)
+    count_data <- filter_data(count_data, exp_metadata, params$nmr_threshold)
```
Comment (mattjmeier) [inline scripts/data_functions.R:75]: What changed here? Just the order?

**Comment by mattjmeier** [comment]
Too bad - I don't think that should stop us from merging this if you think it's good enough for me to test and it doesn't interfere with the rest of the processing steps. I took a look and it seems good.

**Review by mattjmeier — APPROVED** [review]

**Comment by cookkate** [comment]
Yes because the counts are filtered based on which samples remain in the
metadata after it has been filtered

**Comment by cookkate** [comment]
Yeah it should be merge-able.

**Pull request merged**